app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/FwAppInstallerActivity.java

```diff
@@ -165,13 +191,37 @@ public void onClick(View v) {
                 setInstallEnabled(false);
                 installHandler.onStartInstall(device);
                 GBApplication.deviceService().onInstallApp(uri);
+                mProgressBar.setVisibility(View.VISIBLE);
+                IntentFilter intentFilter = new IntentFilter();
+                intentFilter.addAction("nodomain.freeyourgadget.gadgetbridge.INSTALLATION_FAILED");
+                intentFilter.addAction("nodomain.freeyourgadget.gadgetbridge.INSTALLATION_SUCCESS");
+                registerReceiver(installationResultReceiver, intentFilter);
```

Comment:
Register the receiver to the failed/success installation intent, the receiver updates the views (progress bar/button)